### ISSUE TITLE
Recommended switch algorithm

### ISSUE THREAD

**Issue by larseggert** [dthaler/dctcp#1]
Section 3.1 currently says:
```
   The switch indicates congestion to the end nodes by setting the CE
   codepoint in the IP header as specified in Section 5 of [RFC3168].
   For example, the switch may be configured with a congestion
   threshold.  When a packet arrives at the switch and its queue length
   is greater than the congestion threshold, the switch sets the CE
   codepoint in the packet.  However, the actual algorithm for marking
   congestion is an implementation detail of the switch and will
   generally not be known to the sender and receiver.
```
Can we say anything about a recommended switch marking algorithm or configuration?

**Comment by larseggert** [comment]
Any news?

**Comment by sbens** [comment]
Sorry, I lost track of this. No news -- but I just sent another e-mail to the Bing folks.

**Comment by sbens** [comment]
Heard back from a few folks in Bing. The short answer is that they haven't really used this much, so they don't have any meaningful operational experience to share. However, Dave Maltz (one of the authors of the original DCTCP paper) reminded me that the paper gives a recommendation. In particular, that the marking threshold should be K > (RTT * C)/7 where C is packets/second.

Now, when I wrote the IETF draft, I intentionally didn't read the paper because I wasn't sure about copyright, and I didn't want to risk accidentally plagiarizing something. The entire draft is based on the Windows source code alone. Perhaps, we could just refer people to the paper? It's already one of our references.

Maybe something like:

"The actual algorithm for marking congestion is an implementation detail of the switch and will generally not be known to the sender and receiver. However, some guidance is given in section 3.4 of [DCTCP10]."

**Comment by larseggert** [comment]
Thanks for following up! I think it is fine to repeat that guidance in the draft and cite the paper as the source of the information. When properly cited, there is no copyright issue.

**Comment by larseggert** [comment]
Addressed in 5f9f229
